File: submissions/week-4/Day-1/card_checker/card_checker.py
card_number = input("Enter your card: ").strip()

def card_checker(number):
    n = 0  # next second number
    p = 0 # previous number
    num_sum = 0
    double = []
    numbers = []
    list_num = []
    for i in card_number:
        if i != " ":
            numbers.append(int(i))

    for j in range(len(numbers)):
        if j == 0:
            double.append(numbers[j] * 2)
        p = j + 1
        n = j + 2

        if n < len(numbers) and p < len(numbers):
            double_numbers = numbers[n] * 2
            if double_numbers > 9:
                double_numbers = double_numbers - 9
            double.append(double_numbers)

        if p < len(numbers) and n >= len(numbers):
            list_num.append(numbers[p])
    
    lis_su = list_num + double
    num_sum = sum(list_num + double)

    print(lis_su)
    
    if num_sum % 2 != 0:
        print("This card is not a valid card")
    else:
        print("valid card")


if __name__ == "__main__":
    card_checker(card_number)

File: submissions/Week-4/Day-1/PETER-JUNE/cardchecker.js
function isValidCardNumber(cardNumber) {
  let num = [];
  let strCardNumber = cardNumber.toString();
  for (let i = 0; i < strCardNumber.length; i++) {
    let char = strCardNumber[i];
    if (char >= '0' && char <= '9') {
      num.push(Number(char));
    }
  }

  if (num.length < 13 || num.length > 19) {
    return false;
  }

  let sum = 0;
  let shouldDouble = false;

  for (let i = num.length - 1; i >= 0; i--) {
    let digit = num[i];

    if (shouldDouble) {
      digit *= 2;
      if (digit > 9) {
        digit -= 9;
      }
    }

    sum += digit;
    shouldDouble = !shouldDouble;
  }
  if (sum % 10 === 0) {
    console.log('Valid card number');
    return true;
  } else {
    console.log('Invalid card number');
  }

  return false;
}

isValidCardNumber(5199110741698077);
